### PR TITLE
[Generation] Add the dual version explanation for text and chat completion.

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

- [x] Basic class explain.
- [x] `text_completion` and `chat_completion` explain.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the `lint.sh` to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] Code is well-documented.

### CODE REVIEW

File: llama/generation.py [hunk 1]
```diff
@@ -407,9 +452,14 @@ def chat_completion(
         """  # noqa: D205
         if max_gen_len is None:
             max_gen_len = self.model.params.max_seq_len - 1
+
+        # Since the input dialogs are in the form of messages, we need to encode them into tokens.
```
Comment (NANWOOD) [inline llama/generation.py:456]: check "form of messages".

File: llama/generation.py [hunk 2]
```diff
@@ -451,23 +513,38 @@ def chat_completion(
...
             )
+
+            # Check if the last message in the dialog is from the user.
             assert (
                 dialog[-1]["role"] == "user"
             ), f"Last message must be from user, got {dialog[-1]['role']}"
+
+            # Encode the last user message and append it to the dialog tokens. (Note, `eos=False`)
+            #######################################################################################
+            # Q: Why do we need to encode the last user message? Has it not been processed yet?
```
Comment (NANWOOD) [inline llama/generation.py:524]: 是否考虑缩进？

File: llama/generation.py [hunk 3]
```diff
@@ -24,34 +24,67 @@
...
+    logprobs: List[float]
 
 
 class ChatPrediction(TypedDict, total=False):
-    """ChatPrediction class for chat completion predictions."""
+    """ChatPrediction class for chat completion predictions.
+
+    Attributes:
+        generation (Message): The generated message.
+        tokens (List[str]): List of generated tokens. Not mandatory.
```
Comment (NANWOOD) [inline llama/generation.py:62]: Maybe optional？

File: llama/generation_cn_comment.py [hunk 4]
```diff
@@ -24,39 +24,70 @@
...
     content: str
 
 
 class CompletionPrediction(TypedDict, total=False):
-    """CompletionPrediction class for text completion predictions."""
+    """CompletionPrediction 类, 用于文本补全的预测结果.
+
+    Attributes:
+        generation (str): 生成的文本.
+        tokens (List[str]): 生成的 token 列表. 并不强制要求.
```
Comment (NANWOOD) [inline llama/generation_cn_comment.py:43]: 同英文版本

File: llama/generation_cn_comment.py [hunk 5]
```diff
@@ -76,7 +107,7 @@ def build(
             max_batch_size (int): 推理阶段能接受的最大 batch 大小.
             model_parallel_size (Optional[int], optional): 模型并行的进程数.
                 如果未提供, 则从环境中确定. 默认值为 None.
-            seed (int, optional): 随机种子，用于结果复现. 默认种子为 1.
+            seed (int, optional): 随机种子,用于结果复现. 默认种子为 1.
```
Comment (NANWOOD) [inline llama/generation_cn_comment.py:110]: 为什么去掉空格？

File: llama/generation.py (hunk 1 above)
Comment (keli-wen) [reply llama/generation.py:456]: `in the form of` 是以什么什么形式的意思，应该没问题

File: llama/generation_cn_comment.py [hunk 6]
```diff
@@ -393,36 +424,36 @@ def text_completion(
...
-                sequence. If not provided, it's set to the model's maximum sequence length minus 1.
-            logprobs (bool, optional): Flag indicating whether to compute token log probabilities.
-                Defaults to False.
-            echo (bool, optional): Flag indicating whether to include prompt tokens in the
-                generated output. Defaults to False.
+            prompts (List[str]): 输入的 prompt 文本列表.
+            temperature (float, optional): 控制采样随机性的温度值. 默认为 0.6.
+            top_p (float, optional): nucleus 采样中的 top-p 概率阈值. 默认为 0.9.
+            max_gen_len: 最大生成长度. 如果为 None, 则设置为模型的最大序列长度减 1
+                即 (max_seq_len - 1).
```
Comment (NANWOOD) [inline llama/generation_cn_comment.py:434]: 检查换行

File: llama/generation.py [hunk 7]
```diff
@@ -343,9 +376,17 @@ def text_completion(
         """
         if max_gen_len is None:
             max_gen_len = self.model.params.max_seq_len - 1
+
+        # Encode the prompts and add the beginning-of-sequence token.
+        #######################################################################################
+        # Q: Why do not we add the end-of-sequence token?
```
[the remark this answers is not in the document]
Comment (keli-wen) [reply llama/generation.py:382]: How about: "Why don't we add the EOS token?"

File: llama/generation.py (hunk 2 above)
Comment (keli-wen) [reply llama/generation.py:524]: Which line?

File: llama/generation.py (hunk 3 above)
Comment (keli-wen) [reply llama/generation.py:62]: I do agree.

File: llama/generation_cn_comment.py (hunk 4 above)
Comment (keli-wen) [reply llama/generation_cn_comment.py:43]: Get

File: llama/generation_cn_comment.py (hunk 5 above)
Comment (keli-wen) [reply llama/generation_cn_comment.py:110]: I'll fix it.

File: llama/generation_cn_comment.py (hunk 6 above)
Comment (keli-wen) [reply llama/generation_cn_comment.py:434]: Newline because the limitation of line length.

**Comment by keli-wen** [comment]
@NANWOOD Please kindly review it again.

**Pull request merged**